### DISCUSSION TITLE
Image | Odroid M1

### DISCUSSION THREAD

**Discussion by smittyj** [MichaIng/DietPi#5477] [Image Request]
#### Formal device information
- Device name | Odroid M1
- Official product URL | https://www.hardkernel.com/shop/odroid-m1-with-8gbyte-ram/
- GitHub resource URL (if available) | https://github.com/hardkernel?tab=repositories
- Image download URL | https://wiki.odroid.com/odroid-m1/os_images/ubuntu/ubuntu

https://wiki.odroid.com/odroid-m1/odroid-m1
#### Is the SBC officially supported by the Debian installer?
- ...

#### If not, is a reliable 3rd party Debian image available for this SBC?
- ...

#### If not, are there install instructions for Debian available?

- ...

**Comment by fuzzy** [comment]
The debian installer does not support this SoC as far as I know, however, there is a reliable 3rd party ubuntu based image:
https://wiki.odroid.com/odroid-m1/os_images/ubuntu/ubuntu
(At least reliable in the sense that I've been using it without any major issues for the last week)

**Comment by Joulinar** [comment]
DietPi is not supporting Ubuntu.

**Comment by fuzzy** [comment]
Sure but given the similarities between the two, I would think that using it as a blueprint would be relatively straight forward.

**Reply by Joulinar** [reply]
We require an available Debian base image. Otherwise we are not able to install DietPi on top.

**Comment by MichaIng** [comment]
If we can fetch the kernel and bootloader sources/builds/packages, we could use it to generate a Debian image.

Hmm, RK3568B2 SoC, much slower than Odroid N2(+) but higher price. I wonder who would buy this and why? It has many hardware features, SATA, PCIe etc, maybe this is the reason?

Ah, the 4 GiB model is a little chapter than Odroid N2+ (4 GiB as well), still, the use case for this over N2 or C4 looks very small 🤔. I'll ask for a developer sample.

**Reply by KarelHaruda** [reply]
>It has many hardware features, SATA, PCIe etc, maybe this is the reason?

Bingo, without SATA or M.2 it's all just toys with an SD slot

**Comment by smittyj** [comment]
For my use case 8gb of ram and the PCIe HD slot was the reason for the M1.

**Comment by fuzzy** [comment]
The kernel sources are found here: https://github.com/hardkernel/linux/tree/odroidm1-4.19.y I don't think there are release tarballs, but github will kindly generate a snapshot archive. Kernel is a bit out of date sadly. It uses Petitboot which was interesting, I haven't dug into those sources as of yet, but they should be easy to find.

**Reply by MichaIng** [reply]
I'm definitely not going to publish a new image with any Linux lower than 5.x anymore, but for initial testing it should do 😉. Kinda sad that for a new device, official images are outdated 🤔. I hope Hardkernel ships a new kernel for this SBC at some point, at least for Odroid XU4 they did. But based on experience we need to use mainline kernel sources/builds from Armbian or Debian anyway to keep up support with latest Debian versions, e.g. Bookworm kinda requires Linux 5.4 already, i.e. Hardkernels current Linux implies issues with the Debian version released next summer.

**Reply by fuzzy** [reply]
I too was quite dismayed by the downloadable images and kernel versions. I hadn't gotten around to trying to dist-upgrade as @Vebryn suggests so I'm glad that seems a viable path.

**Comment by Vebryn** [comment]
For info, there is a work in progress concerning Debian 11 installer, cf. https://forum.odroid.com/viewtopic.php?p=349719#p349719

Also, despite old downloadable images (Ubuntu Focal 20.04), we can dist-upgrade to 22.04 and get 5.18 kernel.

```
 21:20:48 $ dl linux-image*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                              Version                        Architecture Description
+++-=================================-==============================-============-=============================================
ii  linux-image-4.19.219-odroid-arm64 4.19.219-202205112209~focal    arm64        Linux 4.19 for ODROID (64-bit ARMv8 machines)
ii  linux-image-5.18.0-odroid-arm64   5.18.0-202205132045~jammy      arm64        Linux 5.18 for 64-bit ARMv8 machines
ii  linux-image-odroidm1              6:2022.04-1+202204230436~jammy arm64        Kernel package for ODROID-M1
```

**Comment by terminalwhoami** [comment]
It looks like there is Armbian support for the M1, does that mean we can get a dietpi image soon?

https://www.armbian.com/odroid-m1/

**Reply by MichaIng** [reply]
It should then be easy to implement at least. But I am currently busy with some other hardware implementations.

**Reply by terminalwhoami** [reply]
This post also has images of debian bookworm and ubuntu jammy for the M1 and M1S.  From what I've seen online, the M1S is just different enough that it needs its own image.

https://forum.odroid.com/viewtopic.php?p=379592#p379592

**Comment by muddiver** [comment]
Hello Dietpi-Team! i wanted to ask if there is an image for The Odroid M1 work in progress? I think there are many devices out there waiting for a Dietpi-Version..  Thanks in advance! Greets, Tom

**Reply by Joulinar** [reply]
Not I'm aware of.

**Comment by MichaIng** [comment]
PR up: #8229

**Reply by MichaIng** [reply]
Test images can be found here: https://dietpi.com/downloads/images/testing/
